src/WebJobs.Script.Grpc/Extensions/ScriptInvocationContextExtensions.cs

```diff
@@ -30,21 +33,43 @@ public static async Task<InvocationRequest> ToRpcInvocationRequest(this ScriptIn
 
             foreach (var input in context.Inputs)
             {
-                TypedData rpcValue = null;
-                if (input.val == null || !rpcValueCache.TryGetValue(input.val, out rpcValue))
+                RpcSharedMemory sharedMem = null;
+                ParameterBinding parameterBinding = null;
+                if (isSharedMemoryDataTransferEnabled)
                 {
-                    rpcValue = await input.val.ToRpc(logger, capabilities);
-                    if (input.val != null)
+                    // Try to transfer this data over shared memory instead of RPC
+                    sharedMem = await input.val.ToRpcSharedMemoryAsync(logger, invocationRequest.InvocationId, sharedMemoryManager);
+                }
+
+                if (sharedMem != null)
```

Comment:
Looks like you can simplify here by removing this if block and moving the logic into the preceding if.

Reply:
Also, why do you skip the rpcValueCache logic for shared memory? That was put in place because sometimes a single invocation will send the same value multiple times in a single request via different binding members.

Reply:
Added the corresponding `rpcValueCache` logic as `sharedMemValueCache`.

For the removing of the `if` block - I don't quite follow. The intention is to try to send over shared memory, and if either that fails (`sharedMem == null) or `isSharedMemoryDataTransferEnabled` is `false` then we resort to the existing rpc transfer logic.